src/model/OWL.plt

```diff
@@ -0,0 +1,547 @@
+    is_restriction(R,only(test:'s', test:'Range1'))
+  )),
+  assert_false((
+    subclass_of(test:'A2',R),
+    is_restriction(R,some(test:'t', test:'B'))
+  )).
+
+test('is_restriction/2 with unbound variables') :-
+  % Left argument is unbound
+  assert_true(is_restriction(_,only(test:'s', test:'Range2'))),
+  assert_true(
+    once((
+      is_restriction(ARestr,only(test:'s', test:'Range2')),
+      subclass_of(test:'A',ARestr)
+    ))
+  ),
+  % Right argument is unbound
+  assert_true((subclass_of(test:'A',R),is_restriction(R,_))),
+  assert_true(
+    once((
+      subclass_of(test:'A',A),
+      is_restriction(A,only(test:'s', test:'Range2'))
+    ))
+  ),
+  % Both arguments are unbound
+  assert_true(is_union_of(_,_)).
+
+test('is_union_of') :-
+  assert_true((
+    subclass_of(test:'EUnion',Union1),
+    is_union_of(Union1,union_of([test:'E1',test:'E2']))
+  )),
+  assert_false((
+    subclass_of(test:'EUnion',Union2),
+    is_union_of(Union2,union_of([test:'E1']))
+  )),
+  assert_false((
+    subclass_of(test:'EUnion',Union3),
+    is_union_of(Union3,union_of([test:'E1',test:'E2',test:'E3']))
+  )),
+  % Negative Case
+  assert_false(is_union_of(test:'A',test:'B')).
+
+test('is_union_of with unbound variables') :-
+  % Left argument is unbound
+  assert_true(is_union_of(_,union_of([test:'E1',test:'E2']))),
+  assert_true(
+    once((
+      is_union_of(AUnion,union_of([test:'E1',test:'E2'])),
+      subclass_of(test:'EUnion',AUnion)
+    ))
+  ),
+  % Right argument is unbound
+  assert_true((subclass_of(test:'EUnion',Union4),is_union_of(Union4,_))),
+  assert_true(
+    once((
+      subclass_of(test:'EUnion',BUnion),
+      is_union_of(BUnion,Union),
+      Union=union_of([test:'E1',test:'E2'])
+    ))
+  ),
+  % Both arguments are unbound
+  assert_true(is_union_of(_,_)).
+
+test('is_intersection_of') :-
+  assert_true((
+    subclass_of(test:'CInter',CInter1),
+    is_intersection_of(CInter1,intersection_of([test:'C1',test:'C2']))
+  )),
+  assert_false((
+    subclass_of(test:'CInter',CInter2),
+    is_intersection_of(CInter2,intersection_of([test:'C1']))
+  )),
+  assert_false((
+    subclass_of(test:'CInter',CInter3),
+    is_intersection_of(CInter3,intersection_of([test:'C1',test:'C2',test:'C3']))
+  )),
+  % Negative Case
+  assert_false(is_intersection_of(test:'A',test:'B')).
+
+test('is_intersection_of with unbound variables') :-
+  % Left argument is unbound
+  assert_true(is_intersection_of(_,intersection_of([test:'C1',test:'C2']))),
+  (
+    (
+      subclass_of(test:'CInter',AInter),
+      is_intersection_of(A,intersection_of([test:'C1',test:'C2']))
+    ) -> 
+    assert_unifies(AInter,A) ; 
+    true
+  ),
+  % Right argument is unbound
+  assert_true((
+    subclass_of(test:'CInter',CInter4),
+    is_intersection_of(CInter4,_)
+  )),
+  (
+    (
+      subclass_of(test:'CInter',CInter5), 
+      is_intersection_of(CInter5,B)
+    ) 
+    -> 
+    assert_unifies(intersection_of([test:'C1',test:'C2']),B) ; 
```

Comment:
it is advised to start the newline with *;* to avoid confusion

    -> assert_unifies(intersection_of([test:'C1',test:'C2']),B)
    ;   true